app/code/community/Riskified/Full/Model/Observer.php

```diff
@@ -392,7 +392,7 @@ public function clearRetriesForOrder(Varien_Event_Observer $observer)
         $order = $observer->getOrder();
 
         // Sanity check
-        if (!$order || !$order->getId()) {
+        if (!$order || !$order instanceof Mage_Sales_Model_Order || !$order->getId()) {
```

Comment:
As I wrote previously, you have requested that every request should be created in order helper in postOrder method. For requests like checkout_create or checkout_denied which are triggered by the observer, the order model does not exists (observer create instance of helper and pass checkout data as array). That is the reason why I am checking if order param is instance of Mage_Sales_Model_Order. Of course we can simply check if this is array or not or avoid checking if this is exactly instance of Mage_Sales_Model_Order, but I do not see a reason to do that, moreover I see advantages of that:
It will work if any other extension will rewrite order model. It will be instance of Mage_Sales_Model_Order due to inheritance chain.
We will avoid problems when we will introduce in future any other data transport type, for example other object for other events.
If code composition is ok then there are no problems like that (logic separation, single responsibilities).